File: Exam2019s/exam-aurelia/src/types/Homeworks/HomeworkDTO.ts
export interface StudentHomeworkDTO {
    subjectId: string;
    homeWorkId: string;
    studentName: string;
    studentCode: string;
    isAccepted: boolean;
    isChecked: boolean;
    grade: number;
}

export interface HomeworkDTO {
    id: string;
    title: string;
    description: string;
    subjectTitle: string;
    subjectCode: string;
    subjectId: string;
    deadline: Date;
    studentHomeWorks: StudentHomeworkDTO[];
}

export interface HomeworkGetDTO {
    id: string;
    title: string;
    description: string;
    subjectTitle: string;
    subjectId: string;
    deadline: Date;
}

export interface HomeworkPostDTO {
    title: string;
    subjectId: string;
    description?: string;
    deadline?: Date;
}

export interface HomeworkPutDTO {
    title: string;
    subjectId: string;
    description?: string;
    deadline?: Date;
    id: string;
}

File: Exam2019s/Exam/PublicApi.v1/HomeWorkDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using BLL.App.DTO;

namespace PublicApi.v1
{
    public class HomeWorkDetailsDTO
    {
        public Guid Id { get; set; } = default!;

        public string Title { get; set; } = default!;
        public string? Description { get; set; }

        public string SubjectTitle { get; set; } = default!;
        public string SubjectCode { get; set; } = default!;
        public Guid SubjectId { get; set; } = default!;

        public DateTime? Deadline { get; set; }
        
        public ICollection<StudentHomeWork> StudentHomeWorks { get; set; }
    }
    
    public class StudentHomeWork
    {
        public Guid StudentSubjectId { get; set; } = default!;
        public Guid SubjectId { get; set; } = default!;
        public Guid HomeWorkId { get; set; } = default!;
        
        public string StudentName { get; set; } = default!;
        public string StudentCode { get; set; } = default!;
        
        public bool IsAccepted { get; set; }
        public bool IsChecked { get; set; }
        
        public int Grade { get; set; }
    }

    public class HomeWorkDTO
    {
        public Guid Id { get; set; } = default!;

        public string Title { get; set; } = default!;
        public string? Description { get; set; }

        public string SubjectTitle { get; set; } = default!;
        public string SubjectCode { get; set; } = default!;
        public Guid SubjectId { get; set; } = default!;

        public DateTime? Deadline { get; set; }
    }

    public class HomeWorkPostDTO
    {
        [Required]
        [MinLength(1)]
        [MaxLength(128)]
        public string Title { get; set; } = default!;

        [Required] public Guid SubjectId { get; set; } = default!;

        [MaxLength(4096)] public string? Description { get; set; }

        [DataType(DataType.DateTime)] public DateTime? Deadline { get; set; }
    }

    public class HomeWorkPutDTO
    {
        [Required] public Guid Id { get; set; } = default!;

        [Required]
        [MinLength(1)]
        [MaxLength(128)]
        public string Title { get; set; } = default!;

        [MaxLength(4096)] public string? Description { get; set; }

        [DataType(DataType.DateTime)] public DateTime? Deadline { get; set; }
    }
}